client/src/main/java/org/apache/celeborn/client/write/PushState.java

```diff
@@ -88,4 +100,21 @@ public boolean addBatchData(String addressPair, PartitionLocation loc, int batch
   public DataBatches takeDataBatches(String addressPair) {
     return batchesMap.remove(addressPair);
   }
+
+  public void addFlightBatches(int batchId, PartitionLocation loc) {
+    String addressPair = Utils.genAddressPair(loc);
+    Set<Integer> batchIdSetPerPair =
+        batchIdPerAddressPair.computeIfAbsent(addressPair, id -> new HashSet<>());
+    batchIdSetPerPair.add(batchId);
+    inFlightBatches.put(batchId, loc);
+  }
+
+  public void removeFlightBatches(int batchId) {
+    PartitionLocation loc = inFlightBatches.remove(batchId);
+    String addressPair = Utils.genAddressPair(loc);
```

Comment:
Celeborn just push data to master location then master location's corresponding worker push data to slave location's worker, so here seems we don't need to check if it's peer is null?